packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -326,6 +326,16 @@ class IosProject extends XcodeBasedProject {
     await _updateGeneratedXcodeConfigIfNeeded();
   }
 
+  String? xcworkspacePath() {
+    final List<FileSystemEntity> contents = hostAppRoot.listSync();
+    for (final FileSystemEntity entity in contents) {
+      if (globals.fs.path.extension(entity.path) == '.xcworkspace' && !globals.fs.path.basename(entity.path).startsWith('.')) {
```

Comment:
Add a comment here since it's a little strange, something like
```suggestion
      // On certain volume types there is sometimes a stray `._Runner.xcworkspace` file.
      // Find the first non-hidden xcworkspace.
      if (globals.fs.path.extension(entity.path) == '.xcworkspace' && !globals.fs.path.basename(entity.path).startsWith('.')) {
```

Reply:
Added in https://github.com/flutter/flutter/pull/114099/commits/c8af19f8b885eab084643322b756bede1c03f014.